app/src/components/pages/collectionsPage/collectionsPage.tsx

```diff
@@ -1,17 +1,116 @@
 "use client";
 import PageLayout from "../../pageLayout/pageLayout";
-import React from "react";
-import SearchResults from "../../search/results";
+import React, { useState, useRef, useEffect } from "react";
 import {
   Box,
   Heading,
   HorizontalRule,
   SearchBar,
+  Menu,
+  Text,
+  Pagination,
 } from "@nypl/design-system-react-components";
+import { usePathname, useRouter } from "next/navigation";
 import { headerBreakpoints } from "../../../utils/breakpoints";
+import { CardsGrid } from "../../grids/cardsGrid";
+import LaneLoading from "../../lane/laneLoading";
+import {
+  displayResults,
+  totalNumPages,
+  createCollectionsQueryStringFromObject,
+} from "../../../utils/utils";
+import type { SyntheticEvent } from "react";
 import { createAdobeAnalyticsPageName } from "@/src/utils/utils";
+import NoResultsFound from "../../results/noResultsFound";
+import {
+  DEFAULT_PAGE_NUM,
+  DEFAULT_COLLECTION_SORT,
+  DEFAULT_SEARCH_TERM,
+} from "@/src/config/constants";
+
+export function CollectionsPage({ data, params }) {
+  const { push } = useRouter();
+  const pathname = usePathname();
+  const [isLoaded, setIsLoaded] = useState(false);
+  let collections = [];
+
+  if (data.headers.message === "Collections retrieved successfully") {
+    collections = Array.isArray(data.collection)
+      ? data.collection
+      : [data.collection];
+  }
+
+  const headingRef = useRef<HTMLHeadingElement>(null);
+
+  // pagination
+  const numFound = data.numFound || data.numResults;
+  const totalPages = totalNumPages(numFound, data.perPage);
+
+  // set defaults
+  const [currentPage, setCurrentPage] = useState<number>(
+    params.page || DEFAULT_PAGE_NUM
+  );
+
+  const [currentSort, setCurrentSort] = useState<string>(
+    params.sort || DEFAULT_COLLECTION_SORT
+  );
+
+  const [currentCollectionKeyword, setCurrentCollectionKeyword] =
+    useState<string>(params.collection_keyword || DEFAULT_SEARCH_TERM);
+
+  const handleLoadingState = async (queryString) => {
```

Comment:
It's great you were able to turn this into a reusable function and use it various places. I'm going to be nitpicky about the name and suggest it is updated to something that explains that it's "performing" the search by routing to the new search URL. It is handling the loading state but in the grand scheme of things, it's minor compared to the search that is going on here.

Reply:
fine